OSDOCS-6084: Adding network policy for network observability (#60139)
openshift/openshift-docs: modules/network-observability-create-network-policy.adoc

// Module included in the following assemblies:

// * networking/network_observability/network-observability-network-policy.adoc

:_content-type: PROCEDURE
[id="network-observability-network-policy_{context}"]

> = Create a network policy for Network Observability

Reviewer: Do we want to stick with a gerund for this module title? eg "Creating" ? as per the guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assemblymodule-titles-and-section-headings

You might need to create a network policy to secure ingress traffic to the 'netobserv' namespace. In the web console, you can create a network policy using the form view.

.Procedure
. Navigate to *Networking* -> *NetworkPolicies*.
. Select the `netobserv` project from the *Project* dropdown menu.
. Name the policy. For this example, the policy name is `allow-ingress`.

> . Click *Add ingress rule* three times to create three ingress rules.

Reviewer: I always get confused on this. Is "Ingress" capitalized? https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#ingress https://redhat-documentation.github.io/supplementary-style-guide/#bare-metal-n aren't 100% clear to me.

Author: I think that if you're referencing a resource, like the Ingress Controller, then you capitalize, but if you're referencing traffic, like ingress or egress traffic, then its not capitalized. In this case I'm talking about traffic, so I think its lowercase and its also lowercase in the web console form that I'm referencing.

. Specify the following in the form:
.. Make the following specifications for the first *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from the same namespace*.
.. Make the following specifications for the second *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from inside the cluster*. 
... Click *+ Add namespace selector*. 
... Add the label, `kubernetes.io/metadata.name`, and the selector, `openshift-console`. 
.. Make the following specifications for the third *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from inside the cluster*.
... Click *+ Add namespace selector*. 
... Add the label, `kubernetes.io/metadata.name`, and the selector, `openshift-monitoring`.

.Verification
. Navigate to *Observe* -> *Network Traffic*. 
. View the *Traffic Flows* tab, or any tab, to verify that the data is displayed.

> . Navigate to *Observe* -> *Dashboards* the NetObserv/Health selection and verify that the flows are being ingested and sent to Loki, which is represented in the first graph.

Reviewer: Does the first portion of this sentence make sense to you? eg: 

 Navigate to *Observe* -> *Dashboards* the NetObserv/Health selection

Author: Oop, no I think I'm missing "Look in", so "Look in the NetObserv/health selection"

// Module included in the following assemblies:

// * networking/network_observability/configuring-operators.adoc

:_content-type: PROCEDURE
[id="network-observability-network-policy_{context}"]
= Secure Network Observability with a network policy using the web console
You may need to create a network policy to secure ingress traffic to the 'netobserv' namespace. Using the web console, you can create a network policy 

> 

Author: @stleerh In your video, you deleted a network policy and then added a new one. Will customers need to delete a pre-existing, default policy and add this new one?

.Procedure
. Navigate to *Networking* -> *NetworkPolicies*.
. Select the `netobserv` project from the *Project* dropdown menu.
. Name the policy. For this example, the policy name is `allow-ingress`.
. Click the *Add ingress rule* button twice to create two ingress rules, and make the following specifications in the form:
.. Make the following specifications for the first *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from inside the cluster*. 
... Click *+ Add namespace selector*. 
... Add the label, `kubernetes.io/metadata.name`, and the selector, `openshift-console`. 
.. Make the following specifications for the second *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from inside the cluster*.
... Click *+ Add namespace selector*. 
... Add the label, `kubernetes.io/metadata.name`, and the selector, `openshift-monitoring`.

.Optional

> You can edit the network policy using the YAML editor in the web console. The YAML file should be like the following sample:

Author: @stleerh for the YAML piece, I know in your video, you created the `allow-ingress` policy, then you edited it in the YAML. In my example, I went ahead and put the `openshift-monitoring` in the initial setup and then as an Optional step, say you can edit the policy using the YAML editor.

// Module included in the following assemblies:

// * networking/network_observability/configuring-operators.adoc

:_content-type: PROCEDURE
[id="network-observability-network-policy_{context}"]
= Secure Network Observability with a network policy using the web console

> You may need to create a network policy to secure ingress traffic to the 'netobserv' namespace. Using the web console, you can create a network policy 

Reviewer 2: ```suggestion
You might need to create a network policy to secure ingress traffic to the 'netobserv' namespace. Using the web console, you can create a network policy 
```

Reviewer 2: Looks like the rest of the sentence is missing here. Or it needs to be something like, "You can create a network policy using the web console using the following procedure:"

.Procedure
. Navigate to *Networking* -> *NetworkPolicies*.
. Select the `netobserv` project from the *Project* dropdown menu.
. Name the policy. For this example, the policy name is `allow-ingress`.

> . Click the *Add ingress rule* button twice to create two ingress rules, and make the following specifications in the form:

Reviewer 3: I don't work with our GUIs a lot, so this might be ignorance, but why "Click the **Add ingress rule** button twice" rather than "Click **Add ingress rule** twice"?

Reviewer 3: "Make the following specifications" or "Specify"?

Author: This is a little challenging but basically you have to click "Add ingress rule" two times to populate two forms that need to be filled in.

.. Make the following specifications for the first *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from inside the cluster*. 
... Click *+ Add namespace selector*. 

> ... Add the label, `kubernetes.io/metadata.name`, and the selector, `openshift-console`. 

Reviewer 3: I want this "Add" to be transitive as I read it. That's just a stylistic preference, though.

.. Make the following specifications for the second *Ingress rule*:
... From the *Add allowed source* dropdown menu, select *Allow pods from inside the cluster*.
... Click *+ Add namespace selector*. 
... Add the label, `kubernetes.io/metadata.name`, and the selector, `openshift-monitoring`.

> .Optional

Reviewer 3: This looks more like a reference module to me than an optional subsection of this module.

> You can edit the network policy using the YAML editor in the web console. The YAML file should be like the following sample:

Reviewer 2: ```suggestion
You can edit the network policy using the YAML editor in the web console. The YAML file should include the same parameters as the following example:
```
Should the file that pops up have the same parameters, or does the user need to be sure to include all of these parameters? Unsure what "should be like" means here.

Reviewer 3: ISG prefers "by using" rather than just "using" in this context.

Author: Good point, I think "include all of these parameters" is more like what I mean.

> You can edit the network policy using the YAML editor in the web console. The YAML file should be like the following sample:

Reviewer 3: Why would I perform this step rather than doing what I just did in the main steps?

Author: Its an alternative, so I guess not really Optional, but Alternative. So I see your point in the previous comment about this. I can move it to another module.

> [id="network-observability-network-policy-sample_{context}"]

Reviewer 3: Looks like a clear signal to make this its own module, IMO. :)

.Sample network policy
[source, yaml]
----
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: lockdown-netobserv
  namespace: netobserv
spec:
  podSelector: {}
  ingress:
    - from:
        - podSelector: {}   <1>
        - podSelector: {}
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: openshift-console
        - podSelector: {}
          namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: openshift-monitoring
  policyTypes:
    - Ingress
status: {}
----

> <1> If the `namespaceSelector` is not specified, the `podSelector` defaults to the specified namespace. In this sample, the namespace is `netobserv`.

Reviewer 2: I wonder if this footnote would be best understood if noted on both parameters?

Author: Good point. Addressed.

.Verification
. Navigate to *Observe* -> *Network Traffic*. 

> . View the *Traffic Flows* tab, or any tab, to verify data is displayed.

Reviewer 2: ```suggestion
. View the *Traffic Flows* tab, or any tab, to verify that data is displayed.
```

> . Navigate to *Observe* -> *Dashboards* the NetObserv/Health selection and verify that the flows are being ingested and sent to Loki, which is represented in the first graph.

Reviewer 2: ```suggestion
. Navigate to *Observe* -> *Dashboards* the NetObserv/Health selection and verify that the flows are being received and sent to Loki. This activity is represented in the first graph.
```
Haven't seen "ingested" so curious if it can just be received. If there's other processing happening, then I can see using "ingested." Also not sure if the activity is shown on the graph, or if Loki is? For your consideration.

Author: Yeah, its not just receiving. "Ingest" is a common term used in Network Observability and Logging. First the flows are ingested and enriched with kubernetes data, then sent to Loki or Kafka for further processing.
